prime-router/src/main/kotlin/fhirengine/engine/FHIRTranslator.kt

```diff
@@ -237,4 +239,23 @@ class FHIRTranslator(
         }
         return this
     }
+}
+
+/**
+ * Encodes a message while avoiding an error when MSH-2 is five characters long
+ *
+ * @return the encoded message as a string
+ */
+fun Message.encodePreserveEncodingChars(): String {
+    // get encoding characters ...
+    val msh = this.get("MSH") as Segment
+    val encCharString = Terser.get(msh, 2, 0, 1, 1)
+    val hasFiveEncodingChars = encCharString == "^~\\&#"
+    if (hasFiveEncodingChars) Terser.set(msh, 2, 0, 1, 1, "^~\\&")
+    var encodedMsg = encode()
+    if (hasFiveEncodingChars) {
+        encodedMsg = encodedMsg.replace("^~\\&", "^~\\&#")
+        Terser.set(msh, 2, 0, 1, 1, "^~\\&#")
```

Comment:
Is this line required?  Or just being careful incase the caller doesn't use the returned `encodedMsg`?

Reply:
Added a comment for clarification.